include/deal.II/matrix_free/face_info.h

```diff
@@ -110,6 +110,12 @@ namespace internal
        */
       unsigned char face_orientation;
 
+      /**
+       * Reference-cell type of the given face: 0 for line or quadrilateral,
+       * 1 for triangle.
+       */
+      unsigned char face_type = 0;
```

Comment:
Can you remove the in-line default definition and check that things are set correctly? One thing I wonder is if we access this field in a typical matrix-free evaluation loop nowadays - I always tried to make metadata like the present `struct` as small as possible to avoid too much superfluous cache line transfer during the mat-vec. We probably do, because we access at least the orientation bit. I would be curious if there is a measurable impact on performance.

Reply:
> One thing I wonder is if we access this field in a typical matrix-free evaluation loop nowadays - I always tried to make metadata like the present struct as small as possible to avoid too much superfluous cache line transfer during the mat-vec. We probably do, because we access at least the orientation bit. I would be curious if there is a measurable impact on performance.

We don't need it for each cell once everything is set up. Then, we only need it for setting up `FEFaceEvaluation` (since the value is the same for the whole cell range). I would suggest to move `face_type` to `FaceInfo` in a follow-up PR.

Reply:
OK